packages/utils/src/common/object-to-string-path.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  FindOptionsRelations,
+  FindOptionsSelect,
+  FindOptionsWhere,
+} from "typeorm"
+import { FindOptionsOrder } from "typeorm/find-options/FindOptionsOrder"
+
+/**
+ * Converts a typeorms structure of find options to an
+ * array of string paths
+ * @example
+ * input: {
+ *   test: {
+ *     test1: true,
+ *     test2: true,
+ *     test3: {
+ *       test4: true
+ *     },
+ *   },
+ *   test2: true
+ * }
+ * output: ['test.test1', 'test.test2', 'test.test3.test4', 'test2']
+ * @param input
+ */
+export function objectToStringPath<TEntity>(
+  input:
+    | FindOptionsWhere<TEntity>
+    | FindOptionsSelect<TEntity>
+    | FindOptionsOrder<TEntity>
+    | FindOptionsRelations<TEntity> = {}
+): (keyof TEntity)[] {
```

Comment:
todo: replace by string[] as it does return more than `(keyof TEntity)[]` since it can be sub relations/fields etc